src/anndata/_core/storage.py

```diff
@@ -84,6 +88,8 @@ def coerce_array(
             value = value.A
         return value
     if isinstance(value, pd.DataFrame):
+        if allow_df:
+            raise_value_error_if_multiindex_columns(value)
```

Comment:
could remove due to no coverage, but not bad for exhaustiveness' sake

Reply:
I went the opposite route and removed the `xxxm` checking to rely on this instead.